File: Library.Core.Web/src/app/view-models/magazine/getMagazineView.ts
export class GetMagazineView {
  constructor(
    public magazines: MagazineGetMagazineViewItem[]
  ) { }
}

export class MagazineGetMagazineViewItem {
  constructor(
    public id: number = 0,
    public name?: string,
    public number: number = 0,
    public yearOfPublishing: number = 0
  ) { }
}

File: Library.Core.ViewModelLayer/ViewModels/MagazineViewModels/GetMagazineView.cs
using System.Collections.Generic;

namespace Library.Core.ViewModelLayer.ViewModels.Magazine
{
    public class GetMagazineView
    {
        public ICollection<MagazineGetMagazineViewItem> Magazines { get; set; }

        public GetMagazineView()
        {
            Magazines = new List<MagazineGetMagazineViewItem>();
        }
    }

    public class MagazineGetMagazineViewItem
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int? Number { get; set; }
        public int? YearOfPublishing { get; set; }

        public MagazineGetMagazineViewItem()
        {

        }
    }
}
